### PR TITLE
[sdbus-cpp] Add new port

### LINKED ISSUE #19025: [New Port Request] sdbus-c++
Library name:
sdbus-c++

Library description:
(From README.md)
sdbus-c++ is a high-level C++ D-Bus library for Linux designed to provide expressive, easy-to-use API in modern C++. It adds another layer of abstraction on top of sd-bus, a nice, fresh C D-Bus implementation by systemd.

Source repository URL:
https://github.com/Kistler-Group/sdbus-cpp/

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by AnticliMaxtic**
Add support for sdbus-cpp https://github.com/Kistler-Group/sdbus-cpp

- #### What does your PR fix?  
  Fixes #19025

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  linux , No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  I think so

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @AnticliMaxtic 

Thanks for adding this port.

The errors on CI are like this, could you please look into this?
```
CMake Error at D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/FindPkgConfig.cmake:70 (find_package_handle_standard_args)
  C:/a/1/s/scripts/buildsystems/vcpkg.cmake:786 (_find_package)
  CMakeLists.txt:18 (find_package)
```
You can get the detailed log here https://dev.azure.com/vcpkg/public/_build/results?buildId=59962&view=artifacts&pathAsName=false&type=publishedArtifacts.

File: ports/sdbus-cpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    REF v0.8.3
+    HEAD_REF master
+    SHA512 bdaccd686aeba9f24284c796fac7d0b6d514365e0b757db91e209e1e15c928d9de8ab1d5f5d21671896b07ea762ab4b7c6a5ce0850b17ad08bacb0f1a668cfb2 
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+      "-D BUILD_TESTS=OFF"
+      "-D BUILD_DOC=OFF"
```
Comment (NancyLi1013) [inline ports/sdbus-cpp/portfile.cmake:13]:
```suggestion
    OPTIONS
        -D BUILD_TESTS=OFF
        -D BUILD_DOC=OFF
```

File: ports/sdbus-cpp/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/sdbus-c++)
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+
```
Comment (NancyLi1013) [inline ports/sdbus-cpp/portfile.cmake:23]:
```suggestion
```

File: ports/sdbus-cpp/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "sdbus-cpp",
+  "version-string": "0.8.3",
```
Comment (NancyLi1013) [inline ports/sdbus-cpp/vcpkg.json:3]:
```suggestion
  "version": "0.8.3",
```
This should be served by [version](https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.md#version).

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Please run 
`vcpkg x-add-version sdbus-cpp --overwrite-version` to update files.

**Comment by LilyWangLL** [comment]
@AnticliMaxtic Thanks for your PR. Could you resolve the conlicts?

**Comment by LilyWangLL** [comment]
Pinging @AnticliMaxtic for response. Is work still being done for this PR?

**Comment by AnticliMaxtic** [comment]
@LilyWangLL, yea, the holidays have just been in the way. I will update for requested changes in the first week of January

**Comment by LilyWangLL** [comment]
> @LilyWangLL, yea, the holidays have just been in the way. I will update for requested changes in the first week of January

OK, have a good holidays~

**Pull request closed without merging**